rts/Game/GameControllerTextInput.cpp

```diff
@@ -212,7 +212,7 @@ bool GameControllerTextInput::HandleChatCommand(int key, const std::string& comm
 
 
 // can only be called by CGame (via ConsumePressedKey)
-bool GameControllerTextInput::HandleEditCommand(int key, const std::string& command) {
+bool GameControllerTextInput::HandleEditCommand(int keyCode, int scanCode, const std::string& command) {
```

Comment:
Just one thing to make sure. In the other functions the key/scancodes are unsigned ints. Since they also sometimes get defaulted by `-1`, this is equivalent to getting there to `UINT_MAX - 1`.

However, since `UINT_MAX-1` is larger than `INT_MAX` (by virtue of unsigned integers having double the amount of positive values), if by any chance suck an `unsigned int` keycode gets implicitly converted to a normal `int` it is undefined behavior (because overflow on signed integers is UB).

If this is the case, or probably just in general, it'd probably be best to have all keycodes and scancodes everywhere be `unsigned int`s if possible.

Reply:
If you want to be brief just `unsigned` also works since it is equivalent to `unsigned int` ;)

Reply:
Can be done, I wanted to avoid altering original signatures as much as possible but if you attest that's the correct approach then I can do it.

Reply:
Alternatively I guess you can make them all `int`s; as long as it's consistent so you know there are not broken implicit conversions happening.

Reply:
Well, I'm up for any choice really, just tell me what seems to make more sense.

Reply:
I'm thinking about it. Size-wise they are the same, so that doesn't really matter. First place I'd look at is how they are stored (if they are not just passed around). If storage is `int`, I'd make them all `int`s, and vice-versa. If storage is also mixed, maybe I'd look at the general first entry points for the numbers. If the initial entry point is `int`, then keep going with `int`, and so on.

I don't think there's necessarily an advantage for one or the other. Ideally `unsigned` would probably make more sense given that scancodes are supposed to be positive, but if we're already using values like `-1` as flags then using an `int` could also tell people that this could happen. So as far as "logic" decisions go, I think it's arbitrary whether to use one or another at this point.

Reply:
The -1 value is just for AI input that does not really pass any scan/keycodes. It will also be used for the Lua Callout for GetActionList where -1 denote you don't want to specify a key/scancode.

Reply:
mmh, SDL values are from an enum. It seems from [here](https://en.cppreference.com/w/c/language/enum) that in C enums are supposed to be representable by `int`. If that's the case then yeah using `int` is the best choice probably.

Reply:
Done